src/month.ts

```diff
@@ -598,6 +599,20 @@ function renderMonthDays(
     let indCol = 0;
     let indRow = 0;
     let ind = 0;
+    let curStreakCount = 0;
+    
+    let streakStartDate = startDate.clone().subtract(1, "days");
+    while(curStreakCount < 28) {
+        let curValue = dataset.getValue(streakStartDate);
+        if(curValue != null && ((thresholdType === ThresholdType.LessThan && curValue < threshold) 
+            || (thresholdType === ThresholdType.GreaterThan && curValue > threshold))) {
+            curStreakCount++;
+            streakStartDate = streakStartDate.subtract(1, "days");
+        } else {
+            break;
+        }
+    }
```

Comment:
Minor nit-pick: My preference would be to follow the Return Early pattern. Something like this (while longer) is easier to follow mentally:
```
   while(curStreakCount < 28) {
        let curValue = dataset.getValue(streakStartDate);
        if (curValue == null || curValue == threshold) break;
        if (thresholdType === ThresholdType.GreaterThan && curValue < threshold) break;
        if (thresholdType === ThresholdType.LessThan && curValue > threshold) break;
        curStreakCount++;
        streakStartDate = streakStartDate.subtract(1, "days");
    }
```